src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -106,11 +98,19 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Public
 
-(defn build-ring-handler
-  "Creates the entire compojure application (all routes and middleware)."
-  [request-handler puppet-version]
-  {:pre [(fn? request-handler)]}
-  (-> (root-routes request-handler)
+(defn root-routes
+  "Creates all of the web routes for the master."
+  [request-handler]
+  (comidi/routes
+    (comidi/context "/v2.0"
+                     (v2_0-routes request-handler))
+    (comidi/context ["/" :environment]
+                     (legacy-routes request-handler))))
+
+(defn wrap-middleware
+  [handler puppet-version]
```

Comment:
Same question here about schema validation here as with the `wrap-middleware` function in `certificate_authority_core`.